src/plugins/input-number/index.ts

```diff
@@ -33,14 +34,21 @@ class HSInputNumber
 			this.el.querySelector('[data-hs-input-number-decrement]') || null;
 
 		this.inputValue = 0;
-		this.minInputValue = null;
-		this.maxInputValue = null;
 		if (this.input) {
 			this.inputValue = !isNaN(parseInt(this.input.value)) ? parseInt(this.input.value) : 0;
-			this.minInputValue = ('hsInputNumberMin' in this.input.dataset) ? parseInt(this.input.dataset.hsInputNumberMin) : null;
-			this.maxInputValue = ('hsInputNumberMax' in this.input.dataset) ? parseInt(this.input.dataset.hsInputNumberMax) : null;
 		}
 
+		const data = this.el.dataset.hsInputNumber;
+		const dataOptions: IInputNumberOptions = data ? JSON.parse(data) : { step: 1 };
+		const concatOptions = {
+			...dataOptions,
+			...options
+		};
+
+		this.minInputValue = ('min' in concatOptions) ? concatOptions.min : null;
```

Comment:
please change `minInputValue` to 0 by default, this is necessary for backward compatibility. In our demos the default value is limited to 0. E.g.
`this.minInputValue = 'min' in concatOptions ? concatOptions.min : 0;`

Reply:
@olegpix 
Changing it is not an issue, but the intention for my changes was to mimic the standard HTML number input.

`<input type="number" />` can per default go below 0 and when people needs there number inputs to be min 0, they have to tell that input to do so.

`<input type="number" min="0" />`, just like you have to use a `max` attribute.

I understand that currently your input doesn't allow it, but isn't it better to mimic the defaults of HTML's own input? :)

Reply:
Yes, on the one hand you're right. On the other hand, our plugin was originally developed with this functionality in mind, and it is likely that some users implemented it on their site with this functionality in mind. And we wouldn't want them to have problems after the update.
Unfortunately, I can't make such a decision on my own, here I need @jahaganiev opinion.
Best regards!

Reply:
Hey @olegpix & @jahaganiev 

I have set `min` to default to 0, if `min` isn't in the data options.

I understand your positions, but I still wanted to mentioned my ideas.
I'll simply set `min` to `null` within my project to allow negative numbers, when this gets released.

I hope that in a future update that allows breaking backward compatibility, this function will mimic HTML's own behavior. (either by min/max/step attributes or the default data options mimic that of the input)
Since people in the future that changes to Preline and remove their own `input[number]` to use HSInputNumber might be confused as I was :)

Thanks for the feedback - Looking forward for the continued development of Preline and more fixes to come.